loadtest/control/simplecontroller/actions.go

```diff
@@ -108,3 +109,100 @@ func (c *SimpleController) viewChannel() control.UserStatus {
 		return control.UserStatus{User: c.user, Info: fmt.Sprintf("channel viewed. result: %v", channelViewResponse.ToJson())}
 	*/
 }
+
+func (c *SimpleController) reload() control.UserStatus {
+	teams, err := c.user.Store().Teams()
+	if err != nil {
+		return c.newErrorStatus(err)
+	}
+	chanId := ""
+	if len(teams) > 0 {
+		// Assuming this is a browser and the user has selected the
+		// first team as the default.
+		t := teams[0]
+		// TODO: This will be populated by GetChannelsForTeamForUser API
+		// which is to be implemented. For now, we assume that the user
+		// already belongs to some channels, and they are stored in the
+		// store's memory.
+		chans, err := c.user.Store().Channels(t.Id)
+		if err != nil {
+			return c.newErrorStatus(err)
+		}
+		for _, ch := range chans {
+			// Assuming the first channel to be the currently
+			// selected one.
+			if chanId == "" {
+				chanId = ch.Id
+			}
+			// Marking the channels as viewed
+			_, err := c.user.ViewChannel(&model.ChannelView{
+				ChannelId:     ch.Id,
+				PrevChannelId: "",
+			})
+			if err != nil {
+				return c.newErrorStatus(err)
+			}
+		}
+	}
```

Comment:
I believe this code should go after we get the preferences because in this action we mainly act based on that info and with that result we decide what to fetch from the API.  
Also I would avoid viewing all channels since that is a behaviour which is not strictly related to this function which should be just about reloading. Viewing the last viewed channel (taken from preferences) or a default should be enough.

Reply:
Yeah, but I did see ViewChannel requests for all channels on a reload. Do you think it was the webapp doing things apart from the reload functionality ?

Reply:
> Yeah, but I did see ViewChannel requests for all channels on a reload. Do you think it was the webapp doing things apart from the reload functionality ?

Usually I believe you need to click on an channel's name in the LHS in order to view it. It seems strange it's being called preemptively. That said we should definitely take some extra [mana](https://docs.mattermost.com/process/training.html#mana) to investigate this so we properly mimic the behaviour.